docs/articles/viewing-epoch-based-metrics.rst

```diff
@@ -0,0 +1,232 @@
+performance over multiple epochs.
+
+In this article, we'll show you how to view epoch-based metric data in the WebUI by reporting an
+epoch metric to the Determined master via the Core API. To do this, we'll define an epoch metric.
+This metric is then used as the X-Axis label in the WebUI.
+
+**Recommended**
+
+-  :doc:`Quickstart for Model Developers <../tutorials/quickstart-mdldev>`
+-  :doc:`Core API User Guide <../model-dev-guide/apis-howto/api-core-ug>`
+
+**********************************
+ Set Up Your Training Environment
+**********************************
+
+To begin, you'll need a Determined cluster. If you are new to Determined AI (Determined), you can
+install the Determined library and start a cluster locally.
+
+-  Ensure you have Docker running and then run the following command:
+
+.. code::
+
+   pip install determined
+
+   # If your machine has GPUs:
+   det deploy local cluster-up
+
+   # If your machine does not have GPUs:
+   det deploy local cluster-up --no-gpu
+
+.. include:: ../_shared/note-pip-install-determined.txt
+
+****************************
+ Step 1: Run the Experiment
+****************************
+
+Before reporting metrics, we'll run our experiment and ensure we can see it in the WebUI.
+
+.. note::
+
+   To follow along with the steps shown in this article, you'll need to download the Core API
+   PyTorch MNIST Tutorial files :download:`core_api_pytorch_mnist.tgz
+   </examples/core_api_pytorch_mnist.tgz>`. You can also find the files by visiting the `Github
+   repository
+   <https://github.com/determined-ai/determined/tree/master/examples/tutorials/core_api_pytorch_mnist>`_.
+
+For this step, we'll use our ``model_def.py`` script and its accompanying ``const.yaml`` experiment
+configuration file.
+
+From the directory containing our files, we'll begin by running the following command:
+
+.. code:: bash
+
+   det e create const.yaml . -f
+
+We don't have any data to plot yet, but we'll visit the Determined WebUI to see that our experiment
+is running.
+
+To do this, we'll navigate to ``http://localhost:8080/``.
+
+To sign in, we'll accept the default determined username, leave the password empty, and click **Sign
+In**.
+
+In the WebUI, we'll select our experiment, and then navigate to the **Logs** tab.
+
+************************
+ Step 2: Report Metrics
+************************
+
+In this section, we'll define our epoch metric. We'll also report training and validation metrics to
+the Determined master. To do this, we'll import Determined, and create a
+:class:`~determined.core.Context` object to allow interaction with the master. Then, we'll pass the
+``core_context`` as an argument into ``main()``, ``train()``, and ``test()`` and modify the function
+headers accordingly.
+
+To follow along, use the ``model_def_metrics.py`` script and its accompanying ``metrics.yaml``
+experiment configuration file.
+
+We'll start by importing Determined:
+
+.. code:: bash
+
+   python3
```

Comment:
Drop the `python3` code snippet.

It doesn't look like you say this anywhere, this snippet is implying that somebody is running the code directly from a python3 interpreter.

That is something which few people will do for anything more than very trivial code examples, and our end users are far more likely to do it in a notebook than in a `python3` interpreter.

Also, in general coding tutorials assume the reader knows how to run code in their chosen environment, so I would say it's non-standard to include this detail anyway.

Reply:
dropped